camel-support/src/main/java/io/kaoto/backend/camel/service/step/parser/camelroute/IntegrationStepParserService.java

```diff
@@ -182,13 +185,14 @@
 
     @Override
     public boolean appliesTo(final String yaml) {
-        String[] kinds = new String[]{"Integration"};
-
-        Pattern pattern = Pattern.compile("(kind:)(.+)(\r\n|\r|\n)", Pattern.CASE_INSENSITIVE);
-        Matcher matcher = pattern.matcher(yaml);
+        Matcher matcher = PATTERN.matcher(yaml);
```

Comment:
@Delawen I think at this stage we can dismiss this alert has it has not changed the behavior that existed so far however I think this `appliesTo` method should accept a `JsonNode` or something similar to avoid going back and forth from string <-> json, applying regexp and what not.